From c0b38fa6a28f923b7414de37dd55ecea902ec8ba Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Wed, 26 Mar 2025 14:42:16 +0100
Subject: [PATCH] backend/native: Fetch a11y manager after parent post init

The a11y is only initialized after the parent's post_init, so don't
fetch it until after.
---
 src/backends/native/meta-backend-native.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/backends/native/meta-backend-native.c b/src/backends/native/meta-backend-native.c
index 8732f3f7d2..d1d9da5db3 100644
--- a/src/backends/native/meta-backend-native.c
+++ b/src/backends/native/meta-backend-native.c
@@ -209,7 +209,7 @@ meta_backend_native_post_init (MetaBackend *backend)
     meta_backend_native_get_instance_private (backend_native);
   MetaMonitorManager *monitor_manager =
     meta_backend_get_monitor_manager (backend);
-  MetaA11yManager *a11y_manager = meta_backend_get_a11y_manager (backend);
+  MetaA11yManager *a11y_manager;
 
   META_BACKEND_CLASS (meta_backend_native_parent_class)->post_init (backend);
 
@@ -220,6 +220,7 @@ meta_backend_native_post_init (MetaBackend *backend)
                             G_CALLBACK (update_viewports), backend);
   update_viewports (backend);
 
+  a11y_manager = meta_backend_get_a11y_manager (backend);
   g_signal_connect_object (a11y_manager,
                            "a11y-modifiers-changed",
                            G_CALLBACK (on_a11y_modifiers_changed),
-- 
2.44.0.501.g19981daefd.dirty

